Mathlib/Data/Fin/Basic.lean

```diff
@@ -2570,14 +2571,14 @@ protected theorem zero_mul [NeZero n] (k : Fin n) : (0 : Fin n) * k = 0 := by
 end Mul
 
 open Qq in
-instance toExpr (n : ℕ) : Lean.ToExpr (Fin n) where
-  toTypeExpr := q(Fin $n)
-  toExpr := match n with
+instance toExpr (n : ℕ) : Lean.ToExpr (Fin n) := Lean.ToExpr.mkQ.{0}
```

Comment:
Without the `{0}` this gives
```
type mismatch
  Fin «$n»
has type
  Type : Type 1
but is expected to have type
  Type 127 : Type 128
  ```

Reply:
Yes, this is the `ToLevel` issue I've alluded to on Zulip.  Depending on the order the instances are synthesized, we can end up searching for an instance of `ToLevel.{?u}`.  And since the succ instances has higher priority than the zero instance, we get `ToLevel.{127}` as a result.  If we'd swap the instance priorities, we'd get `ToLevel.{0}`.  Of course, neither choice is desirable.